common/app/model/dotcomrendering/DotcomRenderingDataModel.scala

```diff
@@ -173,4 +163,248 @@ object DotcomRenderingDataModel {
     val jsValue = Json.toJson(model)
     Json.stringify(withoutNull(jsValue))
   }
+
+  def forInteractive(
+      page: InteractivePage,
+      blocks: APIBlocks,
+      request: RequestHeader,
+      pageType: PageType,
+  ): DotcomRenderingDataModel = {
+    apply(
+      page,
+      request,
+      None,
+      linkedData = Nil, // TODO
+      mainBlock = blocks.main,
+      bodyBlocks = blocks.body.getOrElse(Nil),
+      pageType,
+      page.related.hasStoryPackage,
+      keyEvents = Nil,
+    )
+  }
+
+  def forArticle(
+      page: PageWithStoryPackage, // for now, any non-liveblog page type
+      blocks: APIBlocks,
+      request: RequestHeader,
+      pageType: PageType,
+  ): DotcomRenderingDataModel = {
+    val linkedData = LinkedData.forArticle(
+      article = page.article,
+      baseURL = Configuration.amp.baseUrl,
+      fallbackLogo = Configuration.images.fallbackLogo,
+    )
+
+    apply(
+      page,
+      request,
+      None,
+      linkedData,
+      blocks.main,
+      blocks.body.getOrElse(Nil),
+      pageType,
+      page.related.hasStoryPackage,
+      Nil,
+    )
+  }
+
+  def forLiveblog(
+      page: LiveBlogPage,
+      blocks: APIBlocks,
+      request: RequestHeader,
+      pageType: PageType,
+  ): DotcomRenderingDataModel = {
+    val pagination = page.currentPage.pagination.map(paginationInfo => {
+      Pagination(
+        currentPage = page.currentPage.currentPage.pageNumber,
+        totalPages = paginationInfo.numberOfPages,
+        newest = paginationInfo.newest.map(_.suffix),
+        newer = paginationInfo.newer.map(_.suffix),
+        oldest = paginationInfo.oldest.map(_.suffix),
+        older = paginationInfo.older.map(_.suffix),
+      )
+    })
+
+    val bodyBlocks = DotcomRenderingUtils.blocksForLiveblogPage(page, blocks)
+    val keyEvents =
+      blocks.requestedBodyBlocks
+        .getOrElse(Map.empty[String, Seq[APIBlock]])
+        .getOrElse("body:key-events", Seq.empty[APIBlock])
```

Comment:
Not  sure why but this stumped me for a bit, `Option.getOrElse` followed by `Map.getOrElse`